File: Week03/C++/task1.cpp
#include <iostream>
using namespace std;

int main() {

    int x;
    cout << "9'dan büyük bir sayi giriniz: ";
    cin >> x;

    int step = 0;

    cout << x;

    while (x > 9) {

        int temp = x;
        int toplam = 0;

        while (temp > 0) {
            int kalan = temp % 10;
            toplam += kalan;
            temp /= 10;
        }

        x = toplam;
        step++;

        cout << " -> " << x;
    }

    cout << endl;
    cout << "Final deger: " << x << endl;
    cout << "Toplam adim: " << step << endl;

}

File: Week03/Python/task1.py
print("9 dan büyük bir sayı giriniz")
x=int(input())
step=0
while x>9:
 temp=x
 toplam=0
 while temp>0:
     kalan=temp%10
     toplam +=kalan
     temp //=10
 x=toplam
 step+=1
print("Final value:", x)
print("Total steps:", step)